### PR TITLE
Fix a Issue: global variable mRequest lead to memory leak

### PR DESCRIPTION
**Pull request by wangzhengyi**
I use LeakCanary to test new code， there is no memory leak when request is set to local variable.

### CODE REVIEW

File: ThinDownloadManager/src/main/java/com/thin/downloadmanager/DownloadDispatcher.java [hunk 1]
```diff
@@ -75,25 +74,18 @@
...
-                updateDownloadState(DownloadManager.STATUS_STARTED);
-                executeDownload(mRequest.getUri().toString());
+		 		shouldAllowRedirects = true;
+                Log.v(TAG, "Download initiated for " + request.getDownloadId());
+                updateDownloadState(request, DownloadManager.STATUS_STARTED);
+                executeDownload(request, request.getUri().toString());
     		} catch (InterruptedException e) {
                 // We may have been interrupted because it was time to quit.
                 if (mQuit) {
-                    if(mRequest != null) {
```
Comment (smanikandan14) [inline ThinDownloadManager/src/main/java/com/thin/downloadmanager/DownloadDispatcher.java]: We need this block of code, to cancel the timer and update the download status.

**Comment by smanikandan14** [comment]
Rest of the changes looks good.

File: ThinDownloadManager/src/main/java/com/thin/downloadmanager/DownloadDispatcher.java (hunk 1 above)
Comment (wangzhengyi) [inline ThinDownloadManager/src/main/java/com/thin/downloadmanager/DownloadDispatcher.java]: yes, you are right, i will fix this problem.

Comment (wangzhengyi) [inline ThinDownloadManager/src/main/java/com/thin/downloadmanager/DownloadDispatcher.java]: i also think if user quit this dispatcher,  mTime should always be cancel

**Comment by wangzhengyi** [comment]
I already update change, @smanikandan14 please recheck again.

**Comment by smanikandan14** [comment]
+1

**Pull request merged**